### PR TITLE
Add flake8 to CI/CD

### PR DESCRIPTION
**Pull request by ggravlingen**
#### Reason for change
After the big whitespace cleanup, I figured it'd be a good idea to avoid adding that again. One way of doing that is to use flake8 in the CI/CD.

#### Description of change
This PR:
- Adds flake8 to the dev requirements.
- Runs flake8 as a part of tox.
- Adds flake8 settings that makes it pass for the existing code base.
- Removes trailing new lines for a set of files.

#### Steps to Test
Add whitespace to an arbitrary Python file in the `<repo root>/arelle/` directory and run `flake8 arelle`. You should be seeing this:
```
(.venv) vscode ➜ /workspaces/Arelle (master) $ flake8 arelle
arelle/PackageManager.py:37:41: W291 trailing whitespace
```

My VS code settings cleans up white space automatically so I had to use a text editor to add the white space before testing.

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: setup.cfg
Comment (ggravlingen) [inline setup.cfg:75]: I found one of these in the wild when I was adding type hints. `F524` could be a possible bug?

File: setup.cfg
Comment (sagesmith-wf) [inline setup.cfg:88]: Can the comments go inline with the error code here? I worry that if these change the comments aren't going to get updated in the future

File: setup.cfg
Comment (ggravlingen) [reply setup.cfg:124]: That's odd. What version of Python are you running?

File: setup.cfg
Comment (ggravlingen) [reply setup.cfg:88]: You mean putting the comment above the ignored code?

Comment (sagesmith-wf) [reply setup.cfg:88]: Was thinking maybe something like:

``` python
W605, # invalid escape sequence '\d'
```

Comment (ggravlingen) [reply setup.cfg:88]: That would work

File: setup.cfg
Comment (ggravlingen) [reply setup.cfg:9]: Good catch

File: arelle/plugin/validate/XFsyntax/xf.py
Comment (ggravlingen) [inline arelle/plugin/validate/XFsyntax/xf.py:83]: All of these noqa yielded errors like this:
`arelle/plugin/validate/XFsyntax/xf.py:1907:28: NQA101 "# noqa" has no violations`

**Review by sagesmith-wf — DISMISSED** [review]

**Comment by ggravlingen** [comment]
Please let me rebase before you merge. The commit history became a bit messy here.

**Comment by ggravlingen** [comment]
> I don't think you need to reset. You can use `git rebase -i 057c5125~` to clean up the commit history for this PR.

Thanks! I think I got that right except for the commit message.

**Review by austinmatherne-wk — APPROVED** [review]

**Pull request merged**
